### PR TITLE
#3282 - Detect Assessment Offering Dates Change - Part 3 (e2e tests)

### PR DESCRIPTION
**Pull request by sh16011993**
### As a part of this PR, the following e2e tests for the workers were completed:

AssessmentController(e2e)-workflowWrapUp
    √ Should populate the previousDateChangedReportedAssessment when the e-Cert was generated for the 'Original Assessment' and offering dates changed as a result of the reassessment. 
    √ Should not populate the previousDateChangedReportedAssessment when the e-Cert was generated for the 'Original Assessment' and offering dates didn't change as a result of the reassessment.
    √ Should not populate the previousDateChangedReportedAssessment when the e-Cert is not generated for the 'Original Assessment' and offering dates changed as a result of the reassessment.   
    √ Should populate the previousDateChangedReportedAssessment with the most recent reported reassessment when there are previously reported reassessments.
    √ Should not populate the previousDateChangedReportedAssessment if there are multiple reassessments where an intermediate reassessment has a different offering date but the current one changed the offering date back to the value of the latest reported assessment. 
    √ Should not populate the previousDateChangedReportedAssessment if there are multiple reassessments where an intermediate reassessment has a different offering date but the current one changed the offering dates back to the value of the last generated ecert when there are no report based reported assessments. 

### **Screenshot for the above created e2e tests:**

image

------------------------------------------------------------------------------------------------------------------------------------------

**Note:** I have updated the `workFlowWrapUp` test file to use `restoreAllMocks` rather than the `resetAllMocks` because the `restoreAllMocks` restores the function implementation to the original (non-mocked) implementation whereas the `resetAllMocks` replaces the mock implementation with an empty function, returning undefined. The first behavior is what was needed for each test and hence used `restoreAllMocks`.

image

Reference: https://jestjs.io/docs/mock-function-api#mockfnmockrestore

### CODE REVIEW

File: sources/packages/backend/apps/workers/src/controllers/assessment/_tests_/e2e/assessment.controller-workflowWrapUp.e2e-spec.ts [hunk 1]
```diff
@@ -43,13 +52,15 @@ describe("AssessmentController(e2e)-workflowWrapUp", () => {
     assessmentSequentialProcessingService = nestApplication.get(
       AssessmentSequentialProcessingService,
     );
+    auditUser = systemUsersService.systemUser;
+    now = new Date();
   });
 
   beforeEach(() => {
-    jest.resetAllMocks();
+    jest.restoreAllMocks();
```
Comment (andrepestana-aot) [inline sources/packages/backend/apps/workers/src/controllers/assessment/_tests_/e2e/assessment.controller-workflowWrapUp.e2e-spec.ts:60]: 👍

Comment (sh16011993) [reply sources/packages/backend/apps/workers/src/controllers/assessment/_tests_/e2e/assessment.controller-workflowWrapUp.e2e-spec.ts:60]: Thanks for investigating the issue with me. 😃

File: sources/packages/backend/apps/workers/src/controllers/assessment/_tests_/e2e/assessment.controller-workflowWrapUp.e2e-spec.ts [hunk 2]
```diff
@@ -240,4 +250,531 @@ describe("AssessmentController(e2e)-workflowWrapUp", () => {
...
+          coeStatus: COEStatus.completed,
+          disbursementScheduleStatus: DisbursementScheduleStatus.Sent,
+        },
+      },
+    );
+    const originalAssessmentId = savedApplication.currentAssessment.id;
+    const differentStudyStartDate = new Date(
+      savedApplication.currentAssessment.offering.studyStartDate,
+    );
+    differentStudyStartDate.setDate(differentStudyStartDate.getDate() + 1);
```
Comment (andrepestana-aot) [inline sources/packages/backend/apps/workers/src/controllers/assessment/_tests_/e2e/assessment.controller-workflowWrapUp.e2e-spec.ts:276]: what `differentStudyStartDate.getDate() + 1` was supposed to do? Add one day, one sec, one millisec?

Use `addDays` from date-utils.ts please.

Comment (andrepestana-aot) [reply sources/packages/backend/apps/workers/src/controllers/assessment/_tests_/e2e/assessment.controller-workflowWrapUp.e2e-spec.ts:276]: Suggestion:
`const differentStudyStartDate = getISODateOnlyString(addDays(savedApplication.currentAssessment.offering.studyStartDate, 1));`

File: sources/packages/backend/apps/workers/src/controllers/assessment/_tests_/e2e/assessment.controller-workflowWrapUp.e2e-spec.ts [hunk 3]
```diff
@@ -240,4 +250,531 @@ describe("AssessmentController(e2e)-workflowWrapUp", () => {
...
+      studyStartDate: getISODateOnlyString(differentStudyStartDate),
+    });
+    savedApplication.currentAssessment.studentAssessmentStatus =
+      StudentAssessmentStatus.InProgress;
+    await db.studentAssessment.save(savedApplication.currentAssessment);
+    const workflowData = {
+      studentData: {
+        dependantStatus: "independant",
+      },
+    } as WorkflowData;
```
Comment (andrepestana-aot) [inline sources/packages/backend/apps/workers/src/controllers/assessment/_tests_/e2e/assessment.controller-workflowWrapUp.e2e-spec.ts:287]: I know there were some tests with this already but I did a test here and this is just enough:
`const workflowData = {} as WorkflowData;`
Could you please replace it all across this file? I think that would be better for reading it as the workflow data is not important here.

File: sources/packages/backend/libs/test-utils/src/factories/student-assessment.ts [hunk 4]
```diff
@@ -45,5 +48,8 @@ export function createFakeStudentAssessment(
     StudentAssessmentStatus.Submitted;
   assessment.calculationStartDate =
     options?.initialValue?.calculationStartDate ?? null;
+  assessment.previousDateChangedReportedAssessment =
+    options?.initialValue?.previousDateChangedReportedAssessment ?? null;
```
Comment (andrewsignori-aot) [inline sources/packages/backend/libs/test-utils/src/factories/student-assessment.ts:52]: `previousDateChangedReportedAssessment` should go into the `relations` parameters to follow what the other factories are doing.

File: sources/packages/backend/apps/workers/src/controllers/assessment/_tests_/e2e/assessment.controller-workflowWrapUp.e2e-spec.ts [hunk 5]
```diff
@@ -240,4 +250,531 @@ describe("AssessmentController(e2e)-workflowWrapUp", () => {
...
+    const savedOffering = await db.educationProgramOffering.save(
+      updatedOffering,
+    );
+    const initialValue = {
+      triggerType: AssessmentTriggerType.OfferingChange,
+      assessmentDate,
+      studentAssessmentStatus: StudentAssessmentStatus.Completed,
+    };
+    if (options?.reportedDate) {
+      initialValue["reportedDate"] = options?.reportedDate;
```
Comment (andrewsignori-aot) [inline sources/packages/backend/apps/workers/src/controllers/assessment/_tests_/e2e/assessment.controller-workflowWrapUp.e2e-spec.ts:760]: The properties must be typed.

File: sources/packages/backend/apps/workers/src/controllers/assessment/_tests_/e2e/assessment.controller-workflowWrapUp.e2e-spec.ts [hunk 6]
```diff
@@ -240,4 +250,531 @@ describe("AssessmentController(e2e)-workflowWrapUp", () => {
...
+        savedApplication.currentAssessment.id,
+        workflowData,
+      ),
+    );
+
+    // Asserts
+    expect(FakeWorkerJobResult.getResultType(result)).toBe(
+      MockedZeebeJobResult.Complete,
+    );
+    // Asserts that the student assessment status has changed to completed.
```
Comment (andrepestana-aot) [inline sources/packages/backend/apps/workers/src/controllers/assessment/_tests_/e2e/assessment.controller-workflowWrapUp.e2e-spec.ts:301]: This comment doesn't seem to be right.

File: sources/packages/backend/apps/workers/src/controllers/assessment/_tests_/e2e/assessment.controller-workflowWrapUp.e2e-spec.ts [hunk 7]
```diff
@@ -240,4 +250,531 @@ describe("AssessmentController(e2e)-workflowWrapUp", () => {
...
```
Comment (andrepestana-aot) [inline sources/packages/backend/apps/workers/src/controllers/assessment/_tests_/e2e/assessment.controller-workflowWrapUp.e2e-spec.ts:362]: This comment doesn't seem to be right.

File: sources/packages/backend/apps/workers/src/controllers/assessment/_tests_/e2e/assessment.controller-workflowWrapUp.e2e-spec.ts [hunk 8]
```diff
@@ -240,4 +250,531 @@ describe("AssessmentController(e2e)-workflowWrapUp", () => {
...
```
Comment (andrepestana-aot) [inline sources/packages/backend/apps/workers/src/controllers/assessment/_tests_/e2e/assessment.controller-workflowWrapUp.e2e-spec.ts:422]: This comment doesn't seem to be right.

File: sources/packages/backend/apps/workers/src/controllers/assessment/_tests_/e2e/assessment.controller-workflowWrapUp.e2e-spec.ts [hunk 9]
```diff
@@ -240,4 +250,531 @@ describe("AssessmentController(e2e)-workflowWrapUp", () => {
...
+        createSecondDisbursement: false,
+        firstDisbursementInitialValues: {
+          coeStatus: COEStatus.completed,
+        },
+      },
+    );
+    const differentStudyStartDate = new Date(
+      savedApplication.currentAssessment.offering.studyStartDate,
+    );
+    differentStudyStartDate.setDate(differentStudyStartDate.getDate() + 1);
```
Comment (andrepestana-aot) [inline sources/packages/backend/apps/workers/src/controllers/assessment/_tests_/e2e/assessment.controller-workflowWrapUp.e2e-spec.ts:397]: See: https://github.com/bcgov/SIMS/pull/3344/files#r1619290638

File: sources/packages/backend/apps/workers/src/controllers/assessment/_tests_/e2e/assessment.controller-workflowWrapUp.e2e-spec.ts [hunk 10]
```diff
@@ -240,4 +250,531 @@ describe("AssessmentController(e2e)-workflowWrapUp", () => {
...
+          disbursementScheduleStatus: DisbursementScheduleStatus.Sent,
+        },
+      },
+    );
+    const originalAssessment = savedApplication.currentAssessment;
+    const tomorrow = new Date();
+    tomorrow.setDate(tomorrow.getDate() + 1);
+    const differentStudyStartDate = new Date(
+      savedApplication.currentAssessment.offering.studyStartDate,
+    );
```
Comment (andrepestana-aot) [inline sources/packages/backend/apps/workers/src/controllers/assessment/_tests_/e2e/assessment.controller-workflowWrapUp.e2e-spec.ts:460]: Same as https://github.com/bcgov/SIMS/pull/3344/files#r1619290638

File: sources/packages/backend/apps/workers/src/controllers/assessment/_tests_/e2e/assessment.controller-workflowWrapUp.e2e-spec.ts [hunk 11]
```diff
@@ -240,4 +250,531 @@ describe("AssessmentController(e2e)-workflowWrapUp", () => {
...
+      auditUser,
+      {
+        studyStartDate: getISODateOnlyString(differentStudyStartDate),
+        reportedDate: now,
+        previousDateChangedReportedAssessment: originalAssessment,
+      },
+    );
+    const dayAfterTomorrow = new Date();
+    dayAfterTomorrow.setDate(dayAfterTomorrow.getDate() + 2);
+    differentStudyStartDate.setDate(differentStudyStartDate.getDate() + 1);
```
Comment (andrepestana-aot) [inline sources/packages/backend/apps/workers/src/controllers/assessment/_tests_/e2e/assessment.controller-workflowWrapUp.e2e-spec.ts:474]: Same as https://github.com/bcgov/SIMS/pull/3344/files#r1619290638

File: sources/packages/backend/apps/workers/src/controllers/assessment/_tests_/e2e/assessment.controller-workflowWrapUp.e2e-spec.ts [hunk 12]
```diff
@@ -240,4 +250,531 @@ describe("AssessmentController(e2e)-workflowWrapUp", () => {
...
+      auditUser,
+      {
+        studyStartDate: getISODateOnlyString(differentStudyStartDate),
+        reportedDate: now,
+        previousDateChangedReportedAssessment: firstReassessment,
+      },
+    );
+    const afterDayAfterTomorrow = new Date();
+    afterDayAfterTomorrow.setDate(afterDayAfterTomorrow.getDate() + 3);
+    differentStudyStartDate.setDate(differentStudyStartDate.getDate() + 1);
```
Comment (andrepestana-aot) [inline sources/packages/backend/apps/workers/src/controllers/assessment/_tests_/e2e/assessment.controller-workflowWrapUp.e2e-spec.ts:487]: Same as https://github.com/bcgov/SIMS/pull/3344/files#r1619290638

File: sources/packages/backend/apps/workers/src/controllers/assessment/_tests_/e2e/assessment.controller-workflowWrapUp.e2e-spec.ts [hunk 13]
```diff
@@ -240,4 +250,531 @@ describe("AssessmentController(e2e)-workflowWrapUp", () => {
...
+        createSecondDisbursement: false,
+        firstDisbursementInitialValues: {
+          coeStatus: COEStatus.completed,
+          disbursementScheduleStatus: DisbursementScheduleStatus.Sent,
+        },
+      },
+    );
+    const originalAssessment = savedApplication.currentAssessment;
+    const tomorrow = new Date();
+    tomorrow.setDate(tomorrow.getDate() + 1);
```
Comment (andrepestana-aot) [inline sources/packages/backend/apps/workers/src/controllers/assessment/_tests_/e2e/assessment.controller-workflowWrapUp.e2e-spec.ts:554]: Same as https://github.com/bcgov/SIMS/pull/3344/files#r1619290638

File: sources/packages/backend/apps/workers/src/controllers/assessment/_tests_/e2e/assessment.controller-workflowWrapUp.e2e-spec.ts [hunk 14]
```diff
@@ -240,4 +250,531 @@ describe("AssessmentController(e2e)-workflowWrapUp", () => {
...
+    const differentStudyStartDate = new Date(
+      savedApplication.currentAssessment.offering.studyStartDate,
+    );
+    differentStudyStartDate.setDate(differentStudyStartDate.getDate() + 1);
```
Comment (andrepestana-aot) [inline sources/packages/backend/apps/workers/src/controllers/assessment/_tests_/e2e/assessment.controller-workflowWrapUp.e2e-spec.ts:558]: Same as https://github.com/bcgov/SIMS/pull/3344/files#r1619290638

File: sources/packages/backend/apps/workers/src/controllers/assessment/_tests_/e2e/assessment.controller-workflowWrapUp.e2e-spec.ts [hunk 15]
```diff
@@ -240,4 +250,531 @@ describe("AssessmentController(e2e)-workflowWrapUp", () => {
...
+      {
+        studyStartDate: reportedReassessmentStudyStartDate,
+        studyEndDate: reportedReassessmentStudyEndDate,
+        reportedDate: now,
+        previousDateChangedReportedAssessment: originalAssessment,
+      },
+    );
+    const dayAfterTomorrow = new Date();
+    dayAfterTomorrow.setDate(dayAfterTomorrow.getDate() + 2);
+    differentStudyStartDate.setDate(differentStudyStartDate.getDate() + 1);
```
Comment (andrepestana-aot) [inline sources/packages/backend/apps/workers/src/controllers/assessment/_tests_/e2e/assessment.controller-workflowWrapUp.e2e-spec.ts:578]: Same as https://github.com/bcgov/SIMS/pull/3344/files#r1619290638

File: sources/packages/backend/apps/workers/src/controllers/assessment/_tests_/e2e/assessment.controller-workflowWrapUp.e2e-spec.ts [hunk 16]
```diff
@@ -240,4 +250,531 @@ describe("AssessmentController(e2e)-workflowWrapUp", () => {
...
+    );
+    const originalAssessment = savedApplication.currentAssessment;
+    const originalAssessmentStudyStartDate = getISODateOnlyString(
+      savedApplication.currentAssessment.offering.studyStartDate,
+    );
+    const originalAssessmentStudyEndDate = getISODateOnlyString(
+      savedApplication.currentAssessment.offering.studyEndDate,
+    );
+    const tomorrow = new Date();
+    tomorrow.setDate(tomorrow.getDate() + 1);
```
Comment (andrepestana-aot) [inline sources/packages/backend/apps/workers/src/controllers/assessment/_tests_/e2e/assessment.controller-workflowWrapUp.e2e-spec.ts:662]: Same as https://github.com/bcgov/SIMS/pull/3344/files#r1619290638

File: sources/packages/backend/apps/workers/src/controllers/assessment/_tests_/e2e/assessment.controller-workflowWrapUp.e2e-spec.ts [hunk 17]
```diff
@@ -240,4 +250,531 @@ describe("AssessmentController(e2e)-workflowWrapUp", () => {
...
+    const differentStudyStartDate = new Date(
+      savedApplication.currentAssessment.offering.studyStartDate,
+    );
+    differentStudyStartDate.setDate(differentStudyStartDate.getDate() + 1);
```
Comment (andrepestana-aot) [inline sources/packages/backend/apps/workers/src/controllers/assessment/_tests_/e2e/assessment.controller-workflowWrapUp.e2e-spec.ts:666]: Same as https://github.com/bcgov/SIMS/pull/3344/files#r1619290638

File: sources/packages/backend/apps/workers/src/controllers/assessment/_tests_/e2e/assessment.controller-workflowWrapUp.e2e-spec.ts [hunk 18]
```diff
@@ -240,4 +250,531 @@ describe("AssessmentController(e2e)-workflowWrapUp", () => {
...
+    await saveOfferingAndAssessment(savedApplication, tomorrow, auditUser, {
+      studyStartDate: getISODateOnlyString(differentStudyStartDate),
+      previousDateChangedReportedAssessment: originalAssessment,
+    });
+    const dayAfterTomorrow = new Date();
+    dayAfterTomorrow.setDate(dayAfterTomorrow.getDate() + 2);
+    differentStudyStartDate.setDate(differentStudyStartDate.getDate() + 1);
```
Comment (andrepestana-aot) [inline sources/packages/backend/apps/workers/src/controllers/assessment/_tests_/e2e/assessment.controller-workflowWrapUp.e2e-spec.ts:673]: Same as https://github.com/bcgov/SIMS/pull/3344/files#r1619290638

File: sources/packages/backend/apps/workers/src/controllers/assessment/_tests_/e2e/assessment.controller-workflowWrapUp.e2e-spec.ts [hunk 19]
```diff
@@ -240,4 +250,531 @@ describe("AssessmentController(e2e)-workflowWrapUp", () => {
...
+        savedApplication.currentAssessment.id,
+        workflowData,
+      ),
+    );
+
+    // Asserts
+    expect(FakeWorkerJobResult.getResultType(result)).toBe(
+      MockedZeebeJobResult.Complete,
+    );
+    // Asserts that the student assessment status has changed to completed.
```
Comment (andrepestana-aot) [inline sources/packages/backend/apps/workers/src/controllers/assessment/_tests_/e2e/assessment.controller-workflowWrapUp.e2e-spec.ts:704]: This comment doesn't seem to be right.

File: sources/packages/backend/apps/workers/src/controllers/assessment/_tests_/e2e/assessment.controller-workflowWrapUp.e2e-spec.ts [hunk 20]
```diff
@@ -240,4 +250,531 @@ describe("AssessmentController(e2e)-workflowWrapUp", () => {
...
+        previousDateChangedReportedAssessment: true,
+      },
+      where: { id: savedApplication.currentAssessment.id },
+    });
+    expect(expectedAssessment.previousDateChangedReportedAssessment.id).toBe(
+      originalAssessmentId,
+    );
+  });
+
+  it("Should not populate the previousDateChangedReportedAssessment when the e-Cert was generated for the 'Original Assessment' and offering dates didn't change as a result of the reassessment.", async () => {
```
Comment (andrepestana-aot) [inline sources/packages/backend/apps/workers/src/controllers/assessment/_tests_/e2e/assessment.controller-workflowWrapUp.e2e-spec.ts:318]: Please rewrite based on: https://github.com/bcgov/SIMS/pull/3344/files#r1619329059

File: sources/packages/backend/apps/workers/src/controllers/assessment/_tests_/e2e/assessment.controller-workflowWrapUp.e2e-spec.ts [hunk 21]
```diff
@@ -240,4 +250,531 @@ describe("AssessmentController(e2e)-workflowWrapUp", () => {
...
+      relations: {
+        modifier: true,
+        previousDateChangedReportedAssessment: true,
+      },
+      where: { id: savedApplication.currentAssessment.id },
+    });
+    expect(expectedAssessment.previousDateChangedReportedAssessment).toBeNull();
+  });
+
+  it("Should not populate the previousDateChangedReportedAssessment when the e-Cert is not generated for the 'Original Assessment' and offering dates changed as a result of the reassessment.", async () => {
```
Comment (andrepestana-aot) [inline sources/packages/backend/apps/workers/src/controllers/assessment/_tests_/e2e/assessment.controller-workflowWrapUp.e2e-spec.ts:377]: Please rewrite based on: https://github.com/bcgov/SIMS/pull/3344/files#r1619329059

File: sources/packages/backend/apps/workers/src/controllers/assessment/_tests_/e2e/assessment.controller-workflowWrapUp.e2e-spec.ts [hunk 22]
```diff
@@ -240,4 +250,531 @@ describe("AssessmentController(e2e)-workflowWrapUp", () => {
...
+      relations: {
+        modifier: true,
+        previousDateChangedReportedAssessment: true,
+      },
+      where: { id: savedApplication.currentAssessment.id },
+    });
+    expect(expectedAssessment.previousDateChangedReportedAssessment).toBeNull();
+  });
+
+  it("Should populate the previousDateChangedReportedAssessment with the most recent reported reassessment when there are previously reported reassessments.", async () => {
```
Comment (andrepestana-aot) [inline sources/packages/backend/apps/workers/src/controllers/assessment/_tests_/e2e/assessment.controller-workflowWrapUp.e2e-spec.ts:437]: Please rewrite based on: https://github.com/bcgov/SIMS/pull/3344/files#r1619329059

File: sources/packages/backend/apps/workers/src/controllers/assessment/_tests_/e2e/assessment.controller-workflowWrapUp.e2e-spec.ts [hunk 23]
```diff
@@ -240,4 +250,531 @@ describe("AssessmentController(e2e)-workflowWrapUp", () => {
...
+        previousDateChangedReportedAssessment: true,
+      },
+      where: { id: savedApplication.currentAssessment.id },
+    });
+    expect(expectedAssessment.previousDateChangedReportedAssessment.id).toBe(
+      latestReportedAssessment.id,
+    );
+  });
+
+  it("Should not populate the previousDateChangedReportedAssessment if there are multiple reassessments where an intermediate reassessment has a different offering date but the current one changed the offering date back to the value of the latest reported assessment.", async () => {
```
Comment (andrepestana-aot) [inline sources/packages/backend/apps/workers/src/controllers/assessment/_tests_/e2e/assessment.controller-workflowWrapUp.e2e-spec.ts:534]: Please rewrite based on: https://github.com/bcgov/SIMS/pull/3344/files#r1619329059

File: sources/packages/backend/apps/workers/src/controllers/assessment/_tests_/e2e/assessment.controller-workflowWrapUp.e2e-spec.ts [hunk 24]
```diff
@@ -240,4 +250,531 @@ describe("AssessmentController(e2e)-workflowWrapUp", () => {
...
+      relations: {
+        modifier: true,
+        previousDateChangedReportedAssessment: true,
+      },
+      where: { id: savedApplication.currentAssessment.id },
+    });
+    expect(expectedAssessment.previousDateChangedReportedAssessment).toBeNull();
+  });
+
+  it("Should not populate the previousDateChangedReportedAssessment if there are multiple reassessments where an intermediate reassessment has a different offering date but the current one changed the offering dates back to the value of the last generated ecert when there are no report based reported assessments.", async () => {
```
Comment (andrepestana-aot) [inline sources/packages/backend/apps/workers/src/controllers/assessment/_tests_/e2e/assessment.controller-workflowWrapUp.e2e-spec.ts:636]: Please rewrite based on: https://github.com/bcgov/SIMS/pull/3344/files#r1619329059

File: sources/packages/backend/apps/workers/src/controllers/assessment/_tests_/e2e/assessment.controller-workflowWrapUp.e2e-spec.ts (hunk 3 above)
Comment (andrepestana-aot) [reply sources/packages/backend/apps/workers/src/controllers/assessment/_tests_/e2e/assessment.controller-workflowWrapUp.e2e-spec.ts:287]: Or even better:
```
    // Act
    const result = await assessmentController.workflowWrapUp(
      createFakeWorkflowWrapUpPayload(
        savedApplication.currentAssessment.id,
        {} as WorkflowData,
      ),
    );
```

File: sources/packages/backend/apps/workers/src/controllers/assessment/_tests_/e2e/assessment.controller-workflowWrapUp.e2e-spec.ts (hunk 24 above)
[the remark this answers is not in the document]
Comment (sh16011993) [reply sources/packages/backend/apps/workers/src/controllers/assessment/_tests_/e2e/assessment.controller-workflowWrapUp.e2e-spec.ts:636]: This compares the dates with the original assessment and the other one compares it with another reassessment.

File: sources/packages/backend/apps/workers/src/controllers/assessment/_tests_/e2e/assessment.controller-workflowWrapUp.e2e-spec.ts [hunk 25]
```diff
@@ -240,4 +250,508 @@ describe("AssessmentController(e2e)-workflowWrapUp", () => {
...
+          disbursementScheduleStatus: DisbursementScheduleStatus.Sent,
+        },
+      },
+    );
+    const originalAssessmentId = savedApplication.currentAssessment.id;
+    const differentStudyStartDate = getISODateOnlyString(
+      addDays(1, savedApplication.currentAssessment.offering.studyStartDate),
+    );
+    await saveOfferingAndAssessment(savedApplication, now, auditUser, null, {
+      studyStartDate: getISODateOnlyString(differentStudyStartDate),
```
Comment (andrepestana-aot) [inline sources/packages/backend/apps/workers/src/controllers/assessment/_tests_/e2e/assessment.controller-workflowWrapUp.e2e-spec.ts:277]: Now you can remove the getISODateOnlyString from here.

**Review by andrepestana-aot — APPROVED** [review]
Looks good. Thanks for doing the changes.

**Review by andrewsignori-aot — APPROVED** [review]
Thanks for making the changes, looks good 👍

**Pull request merged**